daemon/volumes.go

```diff
@@ -403,3 +404,36 @@ func copyOwnership(source, destination string) error {
 
 	return os.Chmod(destination, os.FileMode(stat.Mode))
 }
+
+func (container *Container) mountVolumes() error {
+	for dest, source := range container.Volumes {
+		v := container.daemon.volumes.Get(source)
+		if v == nil {
+			return fmt.Errorf("could not find volume for %s:%s, impossible to mount", source, dest)
+		}
+
+		destPath, err := container.getResourcePath(dest)
+		if err != nil {
+			return err
+		}
+
+		if err := mount.Mount(source, destPath, "bind", "rbind,rw"); err != nil {
+			return err
```

Comment:
@cpuguy83 we should probably unmountVolumes in case mountVolumes had an error. Maybe not in mountVolumes itself, maybe just in the error branch in the caller.